### ISSUE TITLE
BERT: Pre-training of Deep Bidirectional Transformers for Language Understanding

### ISSUE THREAD

**Issue by kjh8331267** [kjh8331267/Paper_Review#3]
# Abstract
- Bidirectional Encoder Representations from Transformers
- unlabeled text
- both left and right context
- fine-tuned with just one additional output layer
# 1 Introduction
- Language model pre-training(2015)
- sentence-level tasks
- downstream tasks: feature-based and fine-tuning
    - The feature-based approach
      - ELMO(2018)
    - The fine-tuning approach
      - OpenAI GPT(2018)
        - Generative Pre-trained Transformer
        - a left-to-right architecture
    - major limitation: unidirectional, and this limits the choice of architectures
- we improve the fine-tuning based approaches
- masked language model (MLM)
- next sentence prediction (NSP)
  - text-pair representations
# 2 Related Work
## 2.1 Unsupervised Feature-based Approaches
-  representations of words
   - non-neurall (Brown et al., 1992; Ando and Zhang, 2005; Blitzer et al., 2006)
   - neurall (Mikolov et al., 2013; Pennington et al., 2014) 
- word embeddings (Turian et al., 2010)
- sentence embeddings (Kiros et al., 2015; Logeswaran and Lee, 2018)
- paragraph embeddings (Le and Mikolov, 2014)
- ELMo (Peters et al., 2017, 2018a)
  - context-sensitive features
  - from a left-to-right and a right-to-left language model.
## 2.2 Unsupervised Fine-tuning Approaches
- word embedding parameters from unlabeled text(Collobert and Weston, 2008).
- sentence or document encoders (Daiand Le, 2015; Howard and Ruder, 2018; Radford et al., 2018)
  - which produce contextual token representations
  - 1) pre-trained from unlabeled text
  - 2) fine-tuned for a supervised downstream task
  - advantage: few parameters need to be learned from scratch.
     -  OpenAI GPT (Radford et al., 2018)
  - Left-to-right language model and auto-encoder objectives have been used for pre-training such models
## 2.3 Transfer Learning from Supervised Data
- large datasets
  - natural language inference (Conneau et al., 2017)
  - machine translation (McCann et al., 2017)
- Computer vision research (Deng et al., 2009; Yosinski et al., 2014).
  -  the importance of transfer learning from large pre-trained models
# 3 BERT
![f1](https://github.com/kjh8331267/Paper_Review/assets/15272557/da1cf5e6-f20d-4319-bbdf-99359c5f80c2)

## Model Architecture
- a multi-layer bidirectional Transformer encoder
- model sizes
  - $BERT_{BASE}(L=12, H=768, A=12, Total Parameters=110M)$ 
  - $BERT_{LARGE} (L=24, H=1024, A=16, Total Parameters=340M)$
## Input/Output Representations
-  input: a single sentence and a pair of sentences in One token sequence
  - "sequence": single sentence or two sentences packed together
  - "sentence": an arbitrary span of contiguous text
  - a special tokens
    - [CLS]: classification token
      - the first token of every sequence
    - [SEP]: separation token
      -  separating sentence pairs
- input representation
  - summing the corresponding token, segment, and position embeddings
![f2](https://github.com/kjh8331267/Paper_Review/assets/15272557/d1fc63ce-5d76-4840-b08e-a97b36cd9acd)

## 3.1 Pre-training BERT
- using two unsupervised tasks
### Task 1: Masked LM (MLM)
- 1) mask some percentage of the input tokens at random
- 2) predict those masked tokens.
- mismatch between pre-training and fine-tuning
  - since the [MASK] token does not appear during fine-tuning
  - -> not always replace “masked” words
- chooses 15% of the token positions at random
  - 1) the [MASK] token 80%
  - 2) a random token 10%
  - 3) the unchanged i-th token 10%
### Task 2: Next Sentence Prediction (NSP)
- downstream tasks understanding sentence relationships
  - Question Answering (QA)
  - Natural Language Inference (NLI)
- next sentence prediction task
  - choosing the sentences A and B
    - 50% -> the actual next sentenceA (labeled as IsNext)
    - 50% -> a random sentences (labeled as NotNext)
### Pre-training data
- BooksCorpus (800M words)
- English Wikipedia (2,500M words)
  - only the text passages
  - ignore lists, tables, and headers
- a document-level corpus rather than a shuffled sentence-level corpus
  - in order to extract long contiguous sequences
## 3.2 Fine-tuning BERT
- text pairs
  - common pattern
    - independently encoding text pairs
    - applying bidirectional cross attention
  - BERT
    - encoding a concatenated text pair with self-attention
- 1) plug in the task specific inputs and outputs into BERT
- 2) fine tune all the parameters end-to-end.
![f3](https://github.com/kjh8331267/Paper_Review/assets/15272557/245c450b-75aa-4507-89bb-150bbd723a4b)
![f4](https://github.com/kjh8331267/Paper_Review/assets/15272557/28782da7-280b-47f9-bfc9-b23532c13593)
